src/Microsoft.Framework.Configuration/ConfigurationSection.cs

```diff
@@ -19,6 +20,11 @@ public ConfigurationSection(IList<IConfigurationSource> sources, string key)
                 throw new ArgumentNullException(nameof(sources));
             }
 
+            if (prefix == null)
+            {
+                throw new ArgumentNullException(nameof(prefix));
+            }
+
             if (key == null)
```

Comment:
No. Empty keys should not be alllowed. I will change this condition to `IsNullOrEmpty`